api/v1alpha3/machinehealthcheck_types.go

```diff
@@ -83,6 +83,10 @@ type MachineHealthCheckStatus struct {
 	// total number of healthy machines counted by this machine health check
 	// +kubebuilder:validation:Minimum=0
 	CurrentHealthy int32 `json:"currentHealthy,omitempty"`
+
+	// ObservedGeneration is the latest generation observed by the controller.
+	// +optional
```

Comment:
Should this be no longer optional, since we are not setting omitempty? https://github.com/eBay/Kubernetes/blob/master/docs/devel/api-conventions.md#optional-vs-required